builder/upcloud/driver.go

```diff
@@ -0,0 +1,315 @@
+package upcloud
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/UpCloudLtd/upcloud-go-api/upcloud"
+	"github.com/UpCloudLtd/upcloud-go-api/upcloud/client"
+	"github.com/UpCloudLtd/upcloud-go-api/upcloud/request"
+	"github.com/UpCloudLtd/upcloud-go-api/upcloud/service"
+)
+
+type (
+	Driver interface {
+		CreateServer(string, string) (*upcloud.ServerDetails, error)
+		DeleteServer(string) error
+		StopServer(string) error
+		CreateTemplate(string) (*upcloud.Storage, error)
+		GetStorage() (*upcloud.Storage, error)
+		DeleteTemplate(string) error
+	}
+
+	driver struct {
+		svc    *service.Service
+		config *Config
+	}
+)
+
+func NewDriver(c *Config) Driver {
+	client := client.New(c.Username, c.Password)
+	svc := service.New(client)
+	return &driver{
+		svc:    svc,
+		config: c,
+	}
+}
+
+func (d *driver) CreateServer(storageUuid, sshKeyPublic string) (*upcloud.ServerDetails, error) {
+	// Create server
+	request := d.prepareCreateRequest(storageUuid, sshKeyPublic)
+	response, err := d.svc.CreateServer(request)
+	if err != nil {
+		return nil, fmt.Errorf("Error creating server: %s", err)
+	}
+
+	// Wait for server to start
+	err = d.waitDesiredState(response.UUID, upcloud.ServerStateStarted)
+	if err != nil {
+		return nil, err
+	}
+	return response, nil
+}
+
+func (d *driver) DeleteServer(serverUuid string) error {
+	// get storage to delete it once server deleted
+	storage, err := d.getServerStorage(serverUuid)
```

Comment:
There's `d.DeleteServerAndStorages` available to achieve all this in a single command.

Reply:
Fixed.